velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -291,5 +291,14 @@ TEST_F(DateTimeFunctionsTest, dateSub) {
   EXPECT_EQ(parseDate("5881580-07-11"), dateSub("1969-12-31", kMin));
 }
 
+TEST_F(DateTimeFunctionsTest, dayOfWeek) {
+  const auto day = [&](std::optional<int32_t> date) {
+    return evaluateOnce<int64_t, int32_t>("dayofweek(c0)", {date}, {DATE()});
+  };
+  EXPECT_EQ(std::nullopt, day(std::nullopt));
```

Comment:
Can you also add the test for illegal input such as EXPECT_EQ(std::nullopt, day(-2));

Reply:
Sure